File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/ui/views/MTGCardImageView.java
package com.dbottillo.mtgsearchfree.ui.views;

import android.content.Context;
import android.util.AttributeSet;

public class MTGCardImageView extends android.support.v7.widget.AppCompatImageView {

    public MTGCardImageView(Context context) {
        super(context);
    }

    public MTGCardImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MTGCardImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int height = (int) (getMeasuredWidth() * MTGCardView.RATIO_CARD);
        setMeasuredDimension(getMeasuredWidth(), height);
    }

}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/ui/views/MTGCardImageView.kt
package com.dbottillo.mtgsearchfree.ui.views

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet

class MTGCardImageView : AppCompatImageView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val height = (measuredWidth * RATIO_CARD).toInt()
        setMeasuredDimension(measuredWidth, height)
    }
}
